modules/extensions/src/mask/mask-effect.ts

```diff
@@ -76,6 +76,11 @@ export default class MaskEffect implements Effect {
     const channelMap = this._sortMaskChannels(maskLayers);
     // TODO - support multiple views
     const viewport = viewports[0];
+
+    // Rare case no viewpoint exists (e.g. https://github.com/visgl/deck.gl/issues/8059)
+    if (viewport === undefined) {
```

Comment:
I think the logic is better placed in `DeckRenderer`: https://github.com/visgl/deck.gl/pull/8065/files#diff-c3cbad55ae15018bea81712fe9467de37237b99ef51c3e4c4fa987f40400a6c0R68

could you move it there?

Reply:
Such a logic is already placed in DeckRenderer but seems to not work in the reported case (length of viewports is checked against). I do not know how to integrate it there, as my experience with deck.gl is too limited so far.

Reply:
Hi Felix, I just checked your proposal. It doesn't fix the bug as the error is not originating from DeckRenderer but from `pickObject deck-picker.ts:113`

```
Uncaught TypeError: viewport is undefined
    preRender mask-effect.ts:81
    _drawAndSample deck-picker.ts:520
    _pickClosestObject deck-picker.ts:248
    pickObject deck-picker.ts:113
    _pick deck.ts:703
    _pickAndCallback deck.ts:879
    _onRenderFrame deck.ts:1078
    _renderFrame animation-loop.js:256
    redraw animation-loop.js:154
    _animationFrame animation-loop.js:244
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
    requestAnimationFrame request-animation-frame.js:9
    _requestAnimationFrame animation-loop.js:225
    _animationFrame animation-loop.js:245
mask-effect.ts:81:8
```

Reply:
It can also be fixed by changing `if (!pickableLayers) {` to `if (!pickableLayers || viewports.length===0) {`:

```js
if (!pickableLayers || viewports.length===0) {
      return {
        result: [],
        emptyInfo: getEmptyPickingInfo({viewports, x, y, pixelRatio})
      };
    }
```

Would you prefer that? I can update the PR accordingly.